File: src/71. Simplify Path/main.go
package main

import "strings"

func simplifyPath(path string) string {
	parts := strings.Split(path, "/")
	stack := []string{}

	for _, part := range parts {
		if len(stack) > 0 && part == ".." {
			stack = stack[:len(stack)-1]
		} else if part != "" && part != "." && part != ".." {
			stack = append(stack, part)
		}
	}

	res := "/" + strings.Join(stack, "/")

	return res
}

File: src/71. Simplify Path/main.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        parts = path.split("/")
        stack = []

        for part in parts:
            if len(stack) > 0 and part == "..":
                stack.pop()
            elif part != "" and part != "." and part != "..":
                stack.append(part)

        res = "/" + "/".join(stack)

        return res
